WL-4020 Get public announcements for a non accessible site.

ouit0408 opened the pull request:
For merged announcements, if user has no access to the site get public
messages from the site.Added method 'getMessagesPublic' to fetch all
public messages from the channel.

buckett commented on announcement-tool/tool/src/java/org/sakaiproject/announcement/entityprovider/AnnouncementEntityProviderImpl.java:
```diff
@@ -224,6 +224,11 @@ public String getEntityPrefix() {
+				//user may not have access to view the channel but get all public messages in this channel
+				AnnouncementChannel announcementChannel = (AnnouncementChannel)announcementService.getChannelPublic(channel);
+				if(announcementChannel != null){
+					announcements.addAll(announcementChannel.getMessagesPublic(null, true));
```
This doesn't respect the time parameter that's passed into the method.

ouit0408 committed:
WL-4020 Get public announcements for a non accessible site.

For merged announcements, if user has no access to the site get public
messages from the site.Added method 'getMessagesPublic' to fetch all
public messages from the channel.

ouit0408 commented:
Added time restriction for public messages.

buckett commented on announcement-tool/tool/src/java/org/sakaiproject/announcement/entityprovider/AnnouncementEntityProviderImpl.java:
```diff
@@ -224,6 +225,17 @@ public String getEntityPrefix() {
+					List<Message> publicMessages = announcementChannel.getMessagesPublic(null, true);
+					for(Message message : publicMessages){
+						//Add message only if it is within the time range
+						if(isMessageWithinPastNDays(message, numberOfDaysInThePast)){
```
It also doesn't do the number of announcements (`numberOfAnnouncements`).

ouit0408 replied:
as later in the method trimming of decorated announcements for the list is done

buckett replied:
Ok.
